client/my-sites/plans-grid/components/plan-type-selector/components/interval-type-dropdown.tsx

```diff
@@ -46,17 +47,25 @@ export const IntervalTypeDropdown: React.FunctionComponent< IntervalTypeProps >
 	const selectOptionsList = Object.values( optionsList ).map( ( option ) => ( {
 		key: option.key,
 		name: (
-			<AddOnOption href={ option.url }>
+			<IntervalTypeOption>
 				<span className="name"> { option.name } </span>
 				{ option.discountText ? <span className="discount"> { option.discountText } </span> : null }
-			</AddOnOption>
+			</IntervalTypeOption>
 		),
+		url: option.url,
 	} ) );
 
 	return (
 		<CustomSelectControl
 			className="plan-type-selector__interval-type-dropdown"
 			label=""
+			onChange={ ( {
+				selectedItem,
+			}: {
+				selectedItem: { key: SupportedUrlFriendlyTermType; url: string };
+			} ) => {
+				page( `/start/plans${ selectedItem.url }` );
+			} }
```

Comment:
A couple of issues I think with this (the first two are blockers for me):

1. This looks like something we ought to abstract out of a `plans-grid` component. This event is something that will exist when the interval dropdown is used outside of Calypso (and I'm guessing the action there would not be to send user to `/start/plans[something]`). 
2. Not sure if this is correct within Calypso either. Do we also send users to `/start/plans` when used from within admin (`/plans` page)?
3. This line reads ambiguously to me: `page( '/start/plans${ selectedItem.url }' );`. You cannot attach a `URL` to a URL, so something's weird/off in what/how `selectedItem.url` has been implemented and defined. Something's weirder when they are all empty strings `''` right now.

Reply:
I was at the end of the day yesterday and wanted to post something up for review before signing off, so I wasn't focused on the url, but thanks for chiming in Christos. Everything you bring up are great points to address.

I'll work on them shortly 🙂

Reply:
The most recent set of commits reflect our discussion so far. I added an `onPlanTypeSelectorChange` method to the PlanTypeSelector interface, made generation of the url dynamic to accommodate the `/plans` page, and looked into the `url` being appended to a URL path

I'll be explaining my reasoning soon ( and might make another change or two )

Reply:
> This looks like something we ought to abstract out of a plans-grid component. This event is something that will exist when the interval dropdown is used outside of Calypso (and I'm guessing the action there would not be to send user to /start/plans[something]).

I exposed onChange functionality through a callback we add as props to the `PlanTypeSelector` interface. This seems like a reasonable way to allow consumers to define how they'd handle updates of plan type terms, but open to your thoughts

> Not sure if this is correct within Calypso either. Do we also send users to /start/plans when used from within 
admin (/plans page)?

Definitely not 👍 I've amended this for dynamic url generation

> This line reads ambiguously to me: page( '/start/plans${ selectedItem.url }' );. You cannot "attach a URL to a URL path", so something's weird/off in what/how selectedItem.url has been implemented and defined. Something's weirder when they are all empty strings '' right now.

After looking into this, there is, at least to me, unexpected data returned by the the `useIntervalOptions` hook.  The URL attribute might be either a string of coalesced query params in onboarding Ex. `?flags=onboarding%2Finterval-dropdown&intervalType=2yearly` or a relative url in Calypso admin `/plans` Ex. `/plans/2yearly/jeremyyip124.wordpress.com`

I suppose this is a reflection of the fact that plan type terms are updated differently in onboarding and `/plans`. Not ideal, but not entirely sure if there's a graceful way of handling this without taking a deeper look into actual architectural design of plan types. 🤔

Reply:
> After looking into this, there is, at least to me, unexpected data returned by the the useIntervalOptions hook. The URL attribute might be either a string of coalesced query params in onboarding Ex. ?flags=onboarding%2Finterval-dropdown&intervalType=2yearly or a relative url in Calypso admin /plans Ex. /plans/2yearly/jeremyyip124.wordpress.com

The problem here stems from the very fact that the interval toggle has knowledge of the expected URL structure at the consuming end. This is problematic, more so when that knowledge is only so partial (the consumer defines the route path, then the toggle provides the remaining part for it, whether a query or an actual path).

I'll revisit in a bit after reviewing the changes and see if any action before approving. My immediate impression is we should stop using `option.url`, remove that structure completely from interval toggle.

Also, we really need to make a more conscious effort on how we name things. A partial path or a partial query string is NOT a URL. This seems to be a bit of a recurring issue, this mismatch in semantics and naming.

Reply:
It's a bit past the end of my day. I somewhat understand `The plan type selector can just take displayedIntervalTypes and an onIntervalChange callback.`, so if you have time to elaborate on this, that'd be great. If not, no worries -- I'm sure I can glean the more of the meaining tomorrow when I return to this 🙂

Reply:
@chriskmnds + @jdc91 what I understand from this conversation is that we want to do a few things here:

- Simplify the plan type selector and make it more of a purely presentational component by shifting any url generation concerns to consuming components
- Practically speaking, that means refactoring the `interval-type-dropdown`, and, eventually, the `interval-type-toggle` as well.
  - Move any relative path or query string generation from either plan type selector to Plans Features Main
  - Expose a callback that's invoked on plan type selection. Return simple keys and allow consuming components to decide what to do with the information
  - Bonus points? Align with the refactoring effort by scooting plan type selector UI logic into wpcom-plans-ui data store

I'll start moving in this direction, but if any part of this feels off base, please feel free to chime in 🙂

Reply:
That sounds one hundred percent correct :-)

Reply:
> Practically speaking, that means refactoring the interval-type-dropdown, and, eventually, the interval-type-toggle as well.
> - Move any relative path or query string generation from either plan type selector to Plans Features Main
> - Expose a callback that's invoked on plan type selection. Pass simple keys to callback and allow consuming components to decide what to do with the information
> - Bonus points? Align with the refactoring effort by scooting plan type selector UI logic into wpcom-plans-ui data store

Quick update:

I've focused this PR on updating _**only**_ the plan type dropdown. I will handle the plan type toggle in an immediate follow-up. In doing so, I've made several important changes:
- Path / Query string generation for the plan type selector lives [here](https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/plans-grid/components/plan-type-selector/utils.ts). In the spirit of lifting logic outside of the plan type selector component,  I duplicate this code into a new hook `usePlanTypeDestinationCallback` in Plans Features Main
- The plan type toggle still relies on the [plan type selector utils file](https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/plans-grid/components/plan-type-selector/utils.ts) containing Path / Query string generation logic. I will delete the file once I finish refactoring said toggle
- I delete the `option.url` structure from `useIntervalOptions` because it will now be handled outside of the plan type selector
- I expose the `onPlanTypeSelectorChange` callback to consumers to allow them to decide how they'd like to handle plan type selection. This includes path vs. query param processing